File: 卡码/代码随想录/Python/二叉树/01-二叉树遍历方式/LeetCode_0199.py
#199. 二叉树的右视
from typing import *
from ..TreeNode import *

class Solution:
  def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
    res=[]
    q=[]
    if root!=None:
      q.append(root)
    while len(q)>0:
      size=len(q)
      for i in range(size):
        node=q[0]
        q=q[1:]
        if i==size-1:
          res.append(node.val)
        if node.left!=None:
          q.append(node.left)
        if node.right!=None:
          q.append(node.right)
    return res

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/二叉树/_01_二叉树遍历方式/LeetCode_0199.java
package com.kurumi.leetCode.codeThoughts.二叉树._01_二叉树遍历方式;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

import com.kurumi.leetCode.codeThoughts.entity.TreeNode;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 199. 二叉树的右视
 */
public class LeetCode_0199 {
    static class Solution {
        public List<Integer> rightSideView(TreeNode root) {
            List<Integer> res = new ArrayList<>();
            Queue<TreeNode> q = new LinkedList<>();
            if (root != null)
                q.offer(root);
            int size;
            while (!q.isEmpty()) {
                size = q.size();
                for (int i = 0; i < size; i++) {
                    TreeNode node = q.poll();
                    if (i == size - 1)
                        res.add(node.val);
                    if (node.left != null)
                        q.offer(node.left);
                    if (node.right != null)
                        q.offer(node.right);
                }
            }
            return res;
        }
    }
}
